### ISSUE TITLE
CSS expressions are modified and become invalid during compilation

### ISSUE THREAD

**Issue by olivierpascal** [facebook/stylex#351]
**The problem**

Some CSS values are modified during compilation.

**How to reproduce**

```ts
const styles = stylex.create({
  host: {
    margin: `max(0px, (48px - ${vars.iconSize}) / 2)`,
  },
});
```

is compiled to:

```css
.x1luk75g:not(#\#) {
  margin: max(0,(48px - var(--x16dnrjz))/2);
}
```

Which is incorrect, because 0 must have a unit in px for that expression to be valid.

**Expected behavior**

Compile exactly as asked:

```css
.x1luk75g:not(#\#) {
  margin: max(0px, (48px - var(--x16dnrjz)) / 2);
}
```

---

Also, with no var:

```ts
const styles = stylex.create({
  host: {
    margin: 'max(0px, 28px)',
  },
});

The `margin` css property is purely and simply missing in compiled css.

**Comment by nmn** [comment]
Thanks for the report will investigate.

**Comment by nedjulius** [comment]
not sure about the second example (with no var), but not specifying `px` or any other unit near `0` is valid CSS, because all units at `0` represent the same measurement. examples of using `0` without measurement unit are all over the official MDN documentation, and evaluating an expression like `height: max(0, -21px)` would result in `height: 0`, which is a completely valid CSS.

**Comment by nmn** [comment]
@nedjulius The `0` isn't the problem. The problem is that CSS expects spaces around `/` in expressions. Our CSS minifier is being a bit too aggressive!

**Comment by nmn** [comment]
Did some digging and `0` without the `px` doesn't work as reported. This is possibly a browser bug, but we need to support this pattern anyway.

**Comment by nmn** [comment]
Fixed in 0.5.0
